core-bundle/src/Csp/WysiwygStyleProcessor.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Csp;
+
+class WysiwygStyleProcessor
+{
+    public function __construct(private readonly array $allowedCssProperties)
+    {
+    }
+
+    public function extractStyles(string $htmlFragment): array
+    {
+        // Shortcut for performance reasons
+        // TODO: worth it for a regex?
+        if (!str_contains($htmlFragment, 'style=')) {
+            return [];
+        }
```

Comment:
Not needed AFAIK. The regex engine itself has this shortcut implemented already (if the regex starts with a fixed string).